src/Elasticsearch/Product/CustomFieldUpdater.php

```diff
@@ -56,6 +56,7 @@ public function indexCustomFields(EntityWrittenContainerEvent $containerEvent):
 
         if ($customFieldWrittenEvent !== null) {
             $this->customFieldsCreated($customFieldWrittenEvent);
```

Comment:
I see many parts of this can be simplified, couldn't you use `\Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWrittenContainerEvent::getPrimaryKeysWithPropertyChange` and detect custom fields with have `searchable` change and detect $newCreatedFields based on that?